scheduler/cloudformation/cloudformation.go

```diff
@@ -1,5 +1,554 @@
+	services, err := s.Services(app)
+	if err != nil {
+		return nil, err
+	}
+
+	var arns []*string
+
+	// Find all of the tasks started by the ECS services.
+	for _, serviceArn := range services {
+		id, err := arn.ResourceID(serviceArn)
+		if err != nil {
+			return nil, err
+		}
+
+		var taskArns []*string
+		if err := s.ecs.ListTasksPages(&ecs.ListTasksInput{
+			Cluster:     aws.String(s.Cluster),
+			ServiceName: aws.String(id),
+		}, func(resp *ecs.ListTasksOutput, lastPage bool) bool {
+			taskArns = append(taskArns, resp.TaskArns...)
+			return true
+		}); err != nil {
+			return nil, err
+		}
+
+		if len(taskArns) == 0 {
+			continue
+		}
+
+		arns = append(arns, taskArns...)
+	}
+
+	// Find all of the tasks started by Run.
+	if err := s.ecs.ListTasksPages(&ecs.ListTasksInput{
+		Cluster:   aws.String(s.Cluster),
+		StartedBy: aws.String(app),
+	}, func(resp *ecs.ListTasksOutput, lastPage bool) bool {
+		arns = append(arns, resp.TaskArns...)
+		return true
+	}); err != nil {
+		return nil, err
+	}
+
+	resp, err := s.ecs.DescribeTasks(&ecs.DescribeTasksInput{
+		Cluster: aws.String(s.Cluster),
+		Tasks:   arns,
+	})
+
+	return resp.Tasks, err
+}
+
+// Services returns the names of the map that maps the name of a process to the
+// ARN of the ECS service.
+func (s *Scheduler) Services(appID string) (map[string]string, error) {
+	stackName, err := s.stackName(appID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get a summary of all of the stacks resources.
+	var summaries []*cloudformation.StackResourceSummary
+	if err := s.cloudformation.ListStackResourcesPages(&cloudformation.ListStackResourcesInput{
+		StackName: aws.String(stackName),
+	}, func(p *cloudformation.ListStackResourcesOutput, lastPage bool) bool {
+		summaries = append(summaries, p.StackResourceSummaries...)
+		return true
+	}); err != nil {
+		return nil, err
+	}
+
+	services := make(map[string]string)
+	for _, summary := range summaries {
+		if *summary.ResourceType == ecsServiceType {
+			resp, err := s.cloudformation.DescribeStackResource(&cloudformation.DescribeStackResourceInput{
+				StackName:         aws.String(stackName),
+				LogicalResourceId: summary.LogicalResourceId,
+			})
+			if err != nil {
+				return services, err
+			}
+
+			var meta serviceMetadata
+			if err := json.Unmarshal([]byte(*resp.StackResourceDetail.Metadata), &meta); err != nil {
+				return services, err
+			}
+
+			services[meta.Name] = *resp.StackResourceDetail.PhysicalResourceId
+		}
+	}
+
+	return services, nil
+}
+
+// Stop stops the given ECS task.
+func (s *Scheduler) Stop(ctx context.Context, instanceID string) error {
+	_, err := s.ecs.StopTask(&ecs.StopTaskInput{
+		Cluster: aws.String(s.Cluster),
+		Task:    aws.String(instanceID),
+	})
+	return err
+}
+
+// Scale scales the ECS service for the given process to the desired number of
+// instances.
+func (s *Scheduler) Scale(ctx context.Context, appID string, process string, instances uint) error {
```

Comment:
reading through this it seems like this should handle scaling memory and cpu in the template as well?